RecoLocalTracker/SiPixelRecHits/src/VVIObjF.cc

```diff
@@ -132,6 +133,92 @@ VVIObjF::VVIObjF(float kappa, float beta2, int mode) : mode_(mode) {
 
 }  // VVIObjF
 
+// ***************************************************************************************************************************************
+//! Constructor
+//! Set Vavilov parameters kappa and beta2 and define whether to calculate density fcn or distribution fcn
+//! \param kappa - (input) Vavilov kappa parameter [0.01 (Landau-like) < kappa < 10. (Gaussian-like)]
+// ***************************************************************************************************************************************
+
+VVIObjF::VVIObjF(float kappa) {
+  const float xp[9] = {9.29, 2.47, .89, .36, .15, .07, .03, .02, 0.0};
+  const float xq[7] = {.012, .03, .08, .26, .87, 3.83, 11.0};
+  float h_[7];
+  float q, u, x, c1, c2, c3, c4, d1, h4, h5, h6, q2, x1, d, ll, ul, xf1, xf2, rv;
+  int lp, lq, k, l, n;
+
+  // Make sure that the inputs are reasonable
+
+  if (kappa < 0.01f)
+    kappa = 0.01f;
+  if (kappa > 10.f)
+    kappa = 10.f;
+
+  float invKappa = 1.f / kappa;
+  h_[4] = 1.f - 0.42278433999999998f + (7.6f * invKappa);
+  h_[5] = 1;
+  h_[6] = 0.f;
+  h4 = -(7.6f * invKappa) - (0.57721566f + 1.f);
+  h5 = vdt::fast_logf(kappa);
+  h6 = invKappa;
+  t0_ = (h4 - h_[4] * h5 - (h_[4] + 1.f) * (vdt::fast_logf(h_[4]) + VVIObjFDetails::expint(h_[4])) +
+         vdt::fast_expf(-h_[4])) /
+        h_[4];
+
+  // Set up limits for the root search
+
+  for (lp = 0; lp < 9; ++lp) {
+    if (kappa >= xp[lp])
+      break;
+  }
+  ll = -float(lp) - 1.5f;
+  for (lq = 0; lq < 7; ++lq) {
+    if (kappa <= xq[lq])
+      break;
+  }
+  ul = lq - 6.5f;
+  auto f2 = [h_](float x) {
+    return h_[4] - x + h_[5] * (vdt::fast_logf(std::abs(x)) + VVIObjFDetails::expint(x)) - h_[6] * vdt::fast_expf(-x);
```

Comment:
h_[6] is identically 0., the last piece `- h_[6] * vdt::fast_expf(-x);` can be removed.